packages/plasma/src/proofs.ts

```diff
@@ -176,3 +191,20 @@ export function getReceiptBytes(receipt: TransactionReceipt): Buffer {
     }),
   ]);
 }
+
+// getStateSyncTxHash returns block's tx hash for state-sync receipt
+// Bor blockchain includes extra receipt/tx for state-sync logs,
+// but it is not included in transactionRoot or receiptRoot. 
+// So, while calculating proof, we have to exclude them.
+//
+// This is derived from block's hash and number
+// state-sync tx hash = keccak256("matic-bor-receipt-" + block.number + block.hash)
+export function getStateSyncTxHash(block: Block): Buffer {
+  return keccak256(
+    Buffer.concat([
+      toBuffer("matic-bor-receipt-"), // prefix for bor receipt
+      setLengthLeft(toBuffer(block.number), 8), // 8 bytes of block number (BigEndian)
```

Comment:
Is this simply `block.number.toString(16).padStart(64, '0')` ?

Reply:
Kind of -- `block.number.toString(16).padStart(16, '0')`